### PR TITLE
Add docker file to pack required external modules

### PR DESCRIPTION
**Pull request by prakanth97**
## Purpose

$title

### CODE REVIEW

File: base/docker/DockerfilePacked [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+# -----------------------------------------------------------------------
+
+# Ballerina version.
+ARG BALLERINA_VERSION
+
+FROM ballerina/ballerina:${BALLERINA_VERSION}
+
+USER root
+
+RUN bal pull ballerinax/postgresql \
```
Comment (anuruddhal) [inline base/docker/DockerfilePacked:24]: Shall we add SQL drivers as well?

Comment (azinneera) [reply base/docker/DockerfilePacked:24]: bal pull will not pull the latest version of dependencies. We should add null imports in a ball file and run a ball build to get the latest compatible versions of dependencies.

Comment (prakanth97) [reply base/docker/DockerfilePacked:24]: Importing the null import also internally calls the bal pull is not it?

Comment (prakanth97) [reply base/docker/DockerfilePacked]: Updated https://github.com/ballerina-platform/module-ballerina-docker/pull/1051/commits/26106567630adc08e9b042f98e18d18014aece2a

File: base/docker/extension/Ballerina.toml [hunk 2]
```diff
@@ -0,0 +1,8 @@
+[package]
+org = "ballerina"
+name = "extension"
+version = "0.1.0"
+distribution = "2201.9.2"
+
+[build-options]
+observabilityIncluded = true
```
Comment (azinneera) [inline base/docker/extension/Ballerina.toml:8]: Adding a single file would be enough. We don't need the Ballerina.toml and the .gitignore files

File: base/docker/DockerfilePacked [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+
+FROM ballerina/ballerina:${BALLERINA_VERSION}
+
+USER root
+
+# Copy null import ballerina project.
+COPY ./extension /home/ballerina/extension
+WORKDIR /home/ballerina
+
+RUN bal build extension
```
Comment (anuruddhal) [inline base/docker/DockerfilePacked:28]: Let's remove `extension` folder after building.
```
RUN bal build extension && rm -r ./extension
```

File: base/docker/extension/Ballerina.toml (hunk 2 above)
Comment (prakanth97) [reply base/docker/extension/Ballerina.toml]: Updated https://github.com/ballerina-platform/module-ballerina-docker/pull/1051/commits/fef253c19d05fdfd5a58fa591de1b2c46d75f67e

**Review by anuruddhal — APPROVED** [review]

**Pull request merged**